docs/types.rst

```diff
@@ -99,6 +99,15 @@ Address
 -------
 
 ``address``: Holds a 20 byte value (size of an Ethereum address). Address types also have members and serve as a base for all contracts.
+``address payable``: Same as ``address``, but with the additional member ``transfer``.
+
+Implicit conversions from ``address payable`` to ``address`` are allowed, whereas conversions from ``address`` to ``address payable`` are
+not possible (the only way to perform such a conversion is by using an intermediate conversion to ``uint160``).
```

Comment:
This would look like `address x = ...; address payable y = address(uint160(x))`, right? This doesn't make the "payable" too visible, especially if it is used like `f(address(uint160(x)))`...

Reply:
Well, unless we *do* allow ``address payable(x)`` casts, I don't see a way to make it more explicit... Usually such casts shouldn't be used in the first place, though, so I'm not sure that this is a huge problem.

Reply:
I see the main reason for such casts in using other projects' source code that you cannot or do not want to change / update. Basically using code that was not written with addresses you want to send ether to in mind.

Reply:
But perhaps in this case, the best way is to provide a conversion library function or something like that.

Reply:
Maybe we should even add an example for that? Like:

```
contract SomeContract {
    function() payable external {}
}

// Old contract code may return non-payable addresses, even though the
// address actually does have a payable fallback function.
contract OldContract {
    SomeContract c;
    constructor() public { c = new SomeContract(); }
    function actuallyPayableAddress() public view returns (address) {
        return address(c); // address(c) is actually payable!
    }
}

// We can define a library for explicitly converting ``address``
// to ``address payable`` as a workaround.
library address_make_payable {
    function make_payable(address x) internal pure returns (address payable) {
        return address(uint160(x));
    }
}

contract C {
        using address_make_payable for address;
        function f(OldContract c) public payable {
            // The library can be used as an explicit workaround.
            address payable x = c.actuallyPayableAddress().make_payable();
            x.transfer(msg.value);
        }
}
```

Not sure to where to put that, though. Maybe #4892 would be a good place @leonardoalt?

Reply:
We could put it there. It's just a question of whether we want to make the examples short or long.

Reply:
I think short examples in the initial explanation of the change and then another section with only longer examples for each topic may make sense? Longer examples make it more difficult to quickly read through it, but they may help a lot, if the explanation is not entirely clear to someone, so it may still be good to have them somewhere...